javascript/javascript2/week2/Jsweek2.js

```diff
@@ -0,0 +1,183 @@
+         return console.log('Wear sweater.');
+      }
+      else 
+      return console.log('shorts and a t-shirt.');
+ 
+ };
+ clothesToWear = myFunction(0);
+  
+ 
+ 
+  // Event application 
+ 
+   var today = new Date();
+ console.log(today.getDay());
+ let weekDayOfToday=today.getDay();
+ 
+ const weekDays=["sunday","monday","tuseday","wednesday","thursday","friday","saturday"];
+ 
+ function eventDate(x){
+  if(( weekDayOfToday+x)<7){ 
+     console.log(weekDays[ weekDayOfToday+x]);
+     }
+ else ((weekDayOfToday+x)>=7) 
+    /*If sum of weekDayOfToday and event date larger than 7, 
+      use reminder value  to get the value from array weekDays */
+      
+       let rem= (weekDayOfToday+x)%7; 
+         console.log(weekDays[rem]);
+ }
+ eventDate(10);  
+  
+ 
+ 
+  // Student manager
+ 
+   const class07Students = [ ];
+ function addStudentToClass(studentName) {
+    
+   if (studentName=="" ) {
+    console.log(`cannot be  empty`);
+     }
+    else if (class07Students.length >= 6 && studentName=="Queen"){
+       class07Students.push (studentName);
+   }
+    else if (class07Students.length >= 6){
+       console.log("Cannot add more students to class 07")
+    }
+    else if(class07Students.indexOf(studentName) !== -1){  
+        console.log( `Student ${studentName} is already in the class`);
+    }
+     
+        else 
+          return class07Students.push(studentName)
+       }
+ 
+    addStudentToClass ("Mercury");
+    addStudentToClass ("Venus");
+    addStudentToClass ("Earth");
+    addStudentToClass ("Mars");
+    addStudentToClass ("Jupiter");
+    addStudentToClass ("Saturn");
+    addStudentToClass ("Queen");
+ 
+    console.log(class07Students);
+ 
+ 
+  function getNumberOfStudents() {
+    return class07Students.length;
+    } 
+    console.log(getNumberOfStudents()) 
+      
+   
+   // Candy helper 
+   let boughtCandyPrices = [];
+   
+   candyType =["sweet", "chocolate" , "toffee", "chewing-gum"];
+   pricePerWeight=[0.5, 0.7, 1.1, 0.03];
+ 
+   // addCandy function
+ 
+    function addCandy( candyType, weight){
+      if (candyType=="sweet"){
+      return boughtCandyPrices.push(0.5*weight);
+      }
+      else if (candyType=="chocolate"){
+       return boughtCandyPrices.push(0.7*weight);
+       }
+       else if (candyType=="toffee"){
+          return boughtCandyPrices.push(1.1*weight);
+          }
+          else 
+          return boughtCandyPrices.push(0.03*weight);
+    } 
+    // Adding to candy types sweet (40g) and Chocolate(50g)
+ 
+    addCandy( "sweet", 40);
+    addCandy( "chocolate", 50);
+    console.log(boughtCandyPrices) 
+ 
+    
+    const amountToSpend = Math.random() * 100;
+ 
+    let totalPriceCandy=0;
+       for(i=0; i< boughtCandyPrices.length; i++) { 
+          totalPriceCandy+=boughtCandyPrices[i];
+       }
+ 
+      function canBuyMoreCandy(amountToSpend,boughtCandyPrices){
+         
+             if (boughtCandyPrices< amountToSpend ){
```

Comment:
You can also check if it is possible to buy more candies before pushing to the array ie it should be in the `boughtCandyPrices` only if you have money for it (`amountToSpend`)
So the function will use `totalPriceCandy`